### ISSUE TITLE
关于LibriPhrase的训练集

### ISSUE THREAD

**Issue by Lotus2020-tech** [XiYu-uuu/ICASSP2022-ExpResults#1]
大佬，问一下，你在2024年发的TEXT ADAPTIVE DETECTION FOR CUSTOMIZABLE KEYWORD SPOTTING中用到了LibriPhrase，请问你是否有训练集，或者说你是用什么训练的

**Comment by XiYu-uuu** [comment]
Hi, thanks for your attention! 

I didn't use LibriPhrase in this paper, maybe you want to mention another one: "Contrastive Learning With Audio Discrimination For Customizable Keyword Spotting In Continuous Speech"  in ICASSP 2024 (https://arxiv.org/abs/2401.06485), right?

We don't use sement-level training dataset to train our contrastive-learning-based model. Actually, we want to tackle the problem: if the model is trained on segment-level data, it cannot face **co-articulation and streaming segmentation** problems. So we designed some strategies to make our model more robust to these situations with **general ASR dataset: just general LibriSpeech**.

Although our model is designed for streaming-mode KWS, it can also be used in isolated KWS scenario. To verify this, we use **LibriPhrase test part** to test our model. The dataset authors published this part in their github repo. But unfortunately, they don't publish the training part. You are not the first one to ask me this question, and it's actually a great pity that we cannot follow their work becuase we don't have the official training dataset scripts.  I can't help you more in this part. Maybe you can modify the test scripts to generate training dataset.

BTW, I talked with the dataset author at the ICASSP conference, Seoul. He is surprised that there is so much attention to their LibriPhrase, maybe they will publish scripts about training dataset soon.

**Comment by dhyzy2020** [comment]
Yes, what I want to talk about is indeed Contrastive Learning with Audio Discrimination for Customizable Keyword Spotting In Continuous Speech.I would like to take the liberty of asking if you can disclose the code and data for this paper (Contrastive...)?

**Comment by Lotus2020-tech** [comment]
Thank you! Yes, what I want to talk about is indeed Contrastive Learning with Audio Discrimination for Customizable Keyword Spotting In Continuous Speech.I would like to take the liberty of asking if you can disclose the code and data for this paper (Contrastive...)?

**Comment by XiYu-uuu** [comment]
Actually, I just use original LibriSpeech, and splice and augment data in an online mode. There are some hyper-parameters can be configured, such as the number of max/min phonemes in a sliding window, the overlap thresholds for positive or negative samples, etc. So data is not the important part, maybe dataset and dataloader are more important. I am not sure you are still do this work, If you do, you can directly send emails to me. We can have a further discussion through emails or wechat. this is my emaii: yuxi.cs@sjtu.edu.cn

**Comment by Lotus2020-tech** [comment]
I have sent a message to your email address and would very much appreciate a reply, thanks!
